src/libraries/System.Text.RegularExpressions/tests/FunctionalTests/Regex.Match.Tests.cs

```diff
@@ -1048,37 +1048,58 @@ public async Task Match_Timeout_Throws(RegexEngine engine)
             Assert.Throws<RegexMatchTimeoutException>(() => r.Match(input));
         }
 
-        // TODO: Figure out what to do with default timeouts for source generated regexes
+        [ConditionalTheory(typeof(RemoteExecutor), nameof(RemoteExecutor.IsSupported))]
+        [MemberData(nameof(RegexHelpers.AvailableEngines_MemberData), MemberType = typeof(RegexHelpers))]
+        public void Match_InstanceMethods_DefaultTimeout_Throws(RegexEngine engine)
+        {
+            if (RegexHelpers.IsNonBacktracking(engine))
+            {
+                // test relies on backtracking taking a long time
```

Comment:
we appear to have no tests that verify that the non backtracking engine respects the appdomain setting, although it appears it will.